tool/src/java/org/sakaiproject/lessonbuildertool/tool/beans/SimplePageBean.java

```diff
@@ -7346,4 +7354,24 @@ public static String utf8truncate(String input, int length) {
 	return result.toString();
     }
 
+	/**
+	 * Method to add Resources folder into Lessons tool
+	 */
+	public String folderPickerSubmit(){
+		//Check if user has submitted page without selecting folder?
+		String defaultPath = "/group/" + getCurrentSiteId();
+		if(folderPath == null ||  folderPath.equals(defaultPath)){
+			return "failure";
+		}
+		String dataDirectory = defaultPath + "/" + folderPath;
+		contents = "<div data-copyright=\"true\" class=\"no-highlight\" data-description=\"true\" data-directory=" +dataDirectory+ " data-files=\"true\" data-folder-listing=\"true\"></div>";
+		Placement placement = toolManager.getCurrentPlacement();
+		//setting filter false to disable antisamy for folder listing
+		placement.getPlacementConfig().setProperty("filterHtml","false");
```

Comment:
I couldn't see any other uses of this workaround. If you then go to edit this textblock, does the "magic" get stripped out on subsequent saves?